bootstrap/main.py

```diff
@@ -0,0 +1,198 @@
+    '''Make the hidden sentinel that marks a West installation.
+
+    This is similar to .git or .repo directories: it marks the top
+    level directory of the installation. This allows runtime building
+    other paths, etc., which is how this script delegates to the
+    cloned west in wrapper mode (i.e. after 'west init' is run and the
+    "real" west repository has been cloned).'''
+    # The sentinel is a directory in case we ever need to stash
+    # anything inside. It could just as well be an empty file for now,
+    # but this is more future-proof.
+    sentinel = join(directory, WEST_DIR)
+    os.mkdir(sentinel)
+    # On Windows, we have to manually mark a file as hidden.
+    if platform.system() == 'Windows':
+        subprocess.check_call(['attrib', '+H', sentinel])
+
+
+def clone(url, rev, dest):
+    def repository_type(url):
+        if url.startswith(('http:', 'https:', 'git:', 'git+ssh:', 'file:')):
+            return 'GIT'
+        else:
+            return 'UNKNOWN'
+
+    if os.path.exists(dest):
+        msg = 'refusing to clone into existing location {}'.format(dest)
+        raise WestError(msg)
+
+    if repository_type(url) == 'GIT':
+        subprocess.check_call(['git', 'clone', url, '-b', rev, dest])
+    else:
+        raise WestError('Unknown URL scheme for repository: {}'.format(url))
+
+
+#
+# west init
+#
+
+
+def init(argv):
+    '''Command line handler for ``west init`` invocations.
+
+    This exits the program with a nonzero exit code if fatal errors occur.'''
+    init_parser = argparse.ArgumentParser(
+        prog='west init',
+        description='Bootstrap initialize a Zephyr installation')
+    init_parser.add_argument(
+        '-u', '--manifest-url', default=MANIFEST_DEFAULT,
+        help='Zephyr manifest fetch URL, default ' + MANIFEST_DEFAULT)
+    init_parser.add_argument(
+        '--mr', '--manifest-rev', default=MANIFEST_REV_DEFAULT,
+        dest='manifest_rev',
+        help='Manifest revision to fetch, default ' + MANIFEST_REV_DEFAULT)
+    init_parser.add_argument(
+        '-w', '--west-url', default=WEST_DEFAULT,
+        help='West fetch URL, default ' + WEST_DEFAULT)
+    init_parser.add_argument(
+        '--wr', '--west-rev', default=WEST_REV_DEFAULT, dest='west_rev',
+        help='West revision to fetch, default ' + WEST_REV_DEFAULT)
+    init_parser.add_argument(
+        'directory', nargs='?', default=None,
+        help='Initializes in this directory, creating it if necessary')
+
+    args = init_parser.parse_args(args=argv)
+    directory = args.directory or os.getcwd()
+
+    try:
+        topdir = find_west_topdir(directory)
+        init_reinit(topdir, args)
+    except WestNotFound:
+        init_bootstrap(directory, args)
+
+
+def init_bootstrap(directory, args):
+    '''Bootstrap a new manifest + West installation in the given directory.'''
+    if not os.path.isdir(directory):
+        try:
+            os.makedirs(directory, exist_ok=False)
```

Comment:
This creates a folder specified by the user if it doesn't exist, correct? Should we output some text explaining it to the user?